### DISCUSSION TITLE
OpenAI "Which response do you prefer?" fork "renders weirdly"

### DISCUSSION THREAD

**Discussion by stdedos** [danny-avila/LibreChat#2720] [Troubleshooting]
### What happened?

For the same input (https://github.com/danny-avila/LibreChat/issues/2718), the first rendering I get is the "Response 2" with nothing before or after it.

Ofc probably OpenAI is ab-using the system, but I think it'd be nicer if LibreChat:
1. At the very least, "that fork" should've had the previous messages linked
2. "Somehow", it would be "obvious" which fork is the active one, and show that one first (if somehow "both are active", then I have no opinion)

### Steps to Reproduce

1. Export OpenAI data
2. Import them to LibreChat
3. "Find a suitable discussion"
4. Export it

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
I'm not sure what you mean here. When importing conversations with branches, they seem to render fine and I can cycle through as in ChatGPT

https://github.com/danny-avila/LibreChat/assets/110412045/dfff9467-3885-4362-b741-c546d590abbc

It would help to see screenshots/videos of exactly what you mean

**Reply by stdedos** [reply]
![image](https://github.com/danny-avila/LibreChat/assets/133706/a348b284-584c-40fe-ac62-51d125c8bfef)

vs

![img](https://github.com/danny-avila/LibreChat/assets/133706/bb36be26-81d4-4f03-bdeb-2efcbd4c8b9e)

(ik there is a big text block - sorry about that)

Essentially, this is what it looks like in ChatGTP

![image](https://github.com/danny-avila/LibreChat/assets/133706/e86bffdf-d853-46b8-951d-c98ce91b1fa3)

and then, I don't pick either - but I continue texting (is what I do).

What ChatGTP is doing, (I suspect) it is implicitly picks the first response, and moves on

**Reply by stdedos** [reply]
> What ChatGTP is doing, (I suspect) it is implicitly picks the first response, and moves on

... or, idk. It looks weird:

The main thread contains GTP's first response (`Understood! ` ...).

However the fork is created on "my very first message" - i.e.:
1. 1/2 User: Can you help me with that?
2. 2/2 GTP: Understood! ...

Instead, it should've been on a "much later" message, _of the GTP_ (_for my reference: GTP's response to `No, I'm not saying that`_)

(It is complicating, I understand that. I can try to provide an example json - but maybe on a different/anonymized thread, sometime later though)

**Comment by danny-avila** [comment]
an example JSON would help.

the "which response do you prefer" is really just 2 sibling nodes rendered in a unique way via ChatGPT since you never selected one. That's interesting, I didn't know you could leave it un-selected.

In any case, anonymized text for your example would help a lot.

n-ary trees can be very complicated and I'm sure we haven't accounted for every composition.

**Reply by stdedos** [reply]
I managed to find that representation when pressing "the next option". How the chat looks like when refreshed, is this

![image](https://github.com/danny-avila/LibreChat/assets/133706/58c62c45-305f-4d4a-bc20-0199d6846b9c)

So, "I" haven't selected. But I continued chatting.

I know it's annoying, but it will take me some time to come up with that file.
Ofc, if someone has a throwaway chat they can contribute **and** bumps into this, it'd be nice to have more data.
